torch/csrc/autograd/engine.cpp

```diff
@@ -77,6 +82,10 @@ struct GraphTask {
   std::atomic<uint64_t> outstanding_tasks;
   bool keep_graph;
   bool grad_mode;
+  // check whether the user passes inputs for which grad is to be calculated.
+  // if torch.autograd.backward() is called, it is imperative.
+  // if torch.autograd.grad() is called, it is not imperative.
+  bool is_imperative_backward;
```

Comment:
how can we extract the information from `exec_info` about whether used passed the `inputs` that's passed to `torch.autograd.grad`?

Reply:
`exec_info.empty()` means it's `.backward()`, otherwise it's `.grad()`. We should mention that right next to that field in a comment too (can you please add it?)